ociclient/client.go

```diff
@@ -552,3 +565,215 @@ func AddKnownMediaTypesToCtx(ctx context.Context, mediaTypes []string) context.C
 	}
 	return ctx
 }
+
+// *************************************************************************************
+// Docker Manifest v2 Schema 1 Support
+// see also:
+// - https://docs.docker.com/registry/spec/manifest-v2-1/
+// - https://github.com/moby/moby/blob/master/image/v1/imagev1.go
+// - https://github.com/containerd/containerd/blob/main/remotes/docker/schema1/converter.go
+// *************************************************************************************
+
+const (
+	DockerV2Schema1MediaType       = "application/vnd.docker.distribution.manifest.v1+json"
+	DockerV2Schema1SignedMediaType = "application/vnd.docker.distribution.manifest.v1+prettyjws"
+	MediaTypeImageLayerZstd        = "application/vnd.oci.image.layer.v1.tar+zstd"
+)
+
+type fsLayer struct {
+	BlobSum digest.Digest `json:"blobSum"`
+}
+
+type history struct {
+	V1Compatibility string `json:"v1Compatibility"`
+}
+
+type v1Manifest struct {
+	FSLayers []fsLayer `json:"fsLayers"`
+	History  []history `json:"history"`
+}
+
+type v1History struct {
+	Author          string    `json:"author,omitempty"`
+	Created         time.Time `json:"created"`
+	Comment         string    `json:"comment,omitempty"`
+	ThrowAway       *bool     `json:"throwaway,omitempty"`
+	Size            *int      `json:"Size,omitempty"`
+	ContainerConfig struct {
+		Cmd []string `json:"Cmd,omitempty"`
+	} `json:"container_config,omitempty"`
+}
+
+func (c *client) convertV1ManifestToV2(ctx context.Context, ref string, v1ManifestDesc ocispecv1.Descriptor) (ocispecv1.Descriptor, error) {
```

Comment:
I think we should have that function in a separate function that is not a method of `client` for readability.

the file looks already quite big. Can you put all conversion related code in a separate file?